### ISSUE TITLE
DSLX: Array constructor "has_ellipsis" argument could be more useful

### ISSUE THREAD

**Issue by RobSpringer** [google/xls#249]
Array::Array() takes an "has_ellipsis" argument, which indicates that the array was declared with a trailing "...", indicating that the rest of the elements in the array should be filled with the last-specified element.

When creating such an Array, though, the full set of members must be specified, which ultimately renders the argument useless - AFAICT, whenever the property is queried, it's so users can populate the rest of the elements in their view of the array...which is never necessary since the array must be constructed with its full set of members.

Bottom line: either this arg should actually do something (cause the constructor to auto-populate the missing elements? make users have to do that? [I favor the former]) or be eliminated.

**Comment by cdleary** [comment]
I'm not sure there's anything to do here, in the AST the has_ellipsis field is required so that we can auto-format and match the original syntax/text. We always had `ToString()` in that sense, but maybe the autoformatter being more load bearing makes the necessity more clear here.
